src/loadgenerator/sre_recipe_utils.py

```diff
@@ -17,15 +17,99 @@
+        @env.web_ui.app.route("/api/ping")
+        @return_as_json_response
         def ping():
-            resp = make_response(jsonify({"msg": "pong"}), 200)
-            resp.headers["Content-Type"] = 'application/json'
-            return resp
+            return 200, {"msg": "pong"}
+
+        @env.web_ui.app.route("/api/user_count")
+        @return_as_json_response
+        def user_count():
+            """Return the number of currently running users"""
+            return 200, {"user_count": env.runner.user_count}
+
+        @env.web_ui.app.route("/api/spawn/<user_identifier>", methods=['POST'])
+        @return_as_json_response
+        def spawn_by_user_identifier(user_identifier=None):
+            """
+            Spawn a number of users with the SRE Recipe user identifer.
+
+            Form Paramters:
+            - user_count: Required. The total number of users to spawn
+            - spawn_rate: Required. The spawn rate for the users.
+            - stop_after: Optional. If specified, run the load generation only 
+                          for the given number of seconds.
+            """
+            # Required Query Parameters
+            user_count = request.form.get("user_count", default=None, type=int)
+            spawn_rate = request.form.get("spawn_rate", default=None, type=int)
+            user_class = get_sre_recipe_user_class(user_identifier)
+
+            if user_count is None:
+                return 400, {"err": f"Must specify a valid, non-empty, integer value for query parameter 'user_count': {user_count}"}
+            elif spawn_rate is None:
+                return 400, {"err": f"Must specify a valid, non-empty, integer value for query parameter 'spawn_rate': {spawn_rate}"}
+            elif user_count <= 0:
+                return 400, {"err": f"Query parameter 'user_count' must be positive: {user_count}"}
+            elif spawn_rate <= 0:
+                return 400, {"err": f"Query parameter 'spawn_rate' must be positive: {spawn_rate}"}
+            elif user_class is None:
+                return 400, {"err": f"Cannot find SRE Recipe Load for: {user_identifier}"}
+
+            # Optional Query Parameters
+            stop_after = request.form.get("stop_after", default=None, type=int)
+            if stop_after is not None and stop_after <= 0:
+                return 400, {"err": f"Query parameter 'stop_after' must be positive: {stop_after}"}
+
+            # We currently only support running one SRE Recipe load each time
+            # for implementation simplicity.
+            env.runner.quit()  # stop existing load generating users, if any
+            env.user_classes = [user_class]  # replace with the new users
+
+            # wait a short while for all existing users to stop, then
+            # start generating new load with the new user types
+            gevent.spawn_later(WAIT_SECONDS_BEFORE_SPAWN,
+                               lambda: env.runner.start(user_count, spawn_rate))
```

Comment:
Is there a way to watch for existing users to stop, rather than hard-coding a time?

Reply:
Unfortunately, no. There's event hooks when a single user is stopped, but making it work when all users are stopped prob need some state tracking, and can be more convoluted than needed. However, I acknowledge this is not ideal neither. We can also instead do a busy wait loop to fire this only when the number of users are zero. Thoughts?

@daniel-sanche

Reply:
Personally I think the busy wait loop would be better if waiting is important, but I'm not familiar with how locust works. Do we have tests in place that would catch issues around this?

Reply:
We do have tests on this part, but I agree with you, so changed to busy wait with a timeout.